specs/merge/beacon-chain.md

```diff
@@ -198,7 +200,9 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
     process_operations(state, block.body)
     # Pre-merge, skip execution payload processing
     if is_execution_enabled(state, block):
-        process_execution_payload(state, block.body.execution_payload, EXECUTION_ENGINE)  # [New in Merge]
+        # [New in Merge]
+        randao_mix = get_randao_mix(state, get_current_epoch(state))
```

Comment:
The state is already updated with the randao mix, and the below processing function has full access to the state. We can move this line into the `process_execution_payload` I think. Unlike the block body, the state is already part of the processing, so this doesn't really do much for encapsulation, and overriding the latest randao value in the state in a test case to cover possibly different behavior is not too difficult.

Reply:
`process_execution_payload` is designed to be able to run in parallel to the beacon chain block processing function. This is why the source of randao mix depends on the way the function is called; in the spec case it is read from the state, in other cases it might be computed out of block's randao_reveal and previous randao_mix.